app/javascript/components/App.js

```diff
@@ -45,7 +45,12 @@ class App extends Component {
               let myApartments = this.state.apartments.filter(apart => apart.user_id === current_user.id)
               return <ApartmentProtectedIndex apartments={myApartments}/> 
            }}/>
-            <Route path="/apartmentshow" component={ApartmentShow} />
+            <Route path="/apartmentshow/:id" 
+            render= {(props)=>{
+            let id = props.match.params.id
+            let apartment = this.state.apartments.find((apartmentObject)=> apartmentObject.id == id)
+            return <ApartmentShow apartment={apartment} />
```

Comment:
It is not a good practice to use the loose equality. It would be better to convert the id from params to a number.

Reply:
Did you ensure this code works?